packages/util/src/constants.ts

```diff
@@ -14,14 +14,27 @@ export const MAX_INTEGER = new BN(
   16
 )
 
+/**
+ * The max integer that the evm can handle (2^256-1) as a bigint
+ */
+export const MAX_INTEGER_BIGINT = 2n ** 256n - 1n
+
 /**
  * 2^256
+ *
+ * @deprecated
  */
 export const TWO_POW256 = new BN(
   '10000000000000000000000000000000000000000000000000000000000000000',
   16
 )
 
+/**
+ * 2^256
+ */
+export const TWO_POW256_BIGINT =
+  BigInt(0x10000000000000000000000000000000000000000000000000000000000000000)
```

Comment:
cool, could we just do `2n ** 256n` here?

Reply:
Hmm, is this wise if this always have to be recalculated? 🤔

Reply:
that's a good question, if the calculation is run every time the constant is imported or if it is calculated once on init of the package.

To test:

constants.js
```js
const TWO_POW256_BIGINT_HEX =
  BigInt(0x10000000000000000000000000000000000000000000000000000000000000000);

const TWO_POW256_BIGINT_EXP = 2n ** 256n;

exports = [TWO_POW256_BIGINT_HEX, TWO_POW256_BIGINT_EXP];
```

index.js
```js
const Benchmark = require("benchmark");
const BN = require("bn.js");

const suite = new Benchmark.Suite();

suite
  .add("bigint init with hex", function () {
    const { TWO_POW256_BIGINT_HEX } = require("./constants");
    if (100n > TWO_POW256_BIGINT_HEX) return true;
  })
  .add("bigint init with exp", function () {
    const { TWO_POW256_BIGINT_EXP } = require("./constants");
    if (100n > TWO_POW256_BIGINT_EXP) return true;
  })
  .on("cycle", function (event) {
    console.log(String(event.target));
  })
  .on("complete", function () {
    console.log("Fastest is " + this.filter("fastest").map("name"));
  })
  .run({ async: true });
```

On my Macbook Air (M1, 2020) my results are always about even:

```
bigint init with hex x 6,561,811 ops/sec ±0.37% (97 runs sampled)
bigint init with exp x 6,573,137 ops/sec ±0.10% (99 runs sampled)
Fastest is bigint init with exp,bigint init with hex
```

I'm not quite sure if this exactly replicates our util setup, but thought I'd try it.

Let's compare this to inline definitions and not importing (snippet):

```js
const TWO_POW256_BIGINT_HEX =
  BigInt(0x10000000000000000000000000000000000000000000000000000000000000000);
const TWO_POW256_BIGINT_EXP = 2n ** 256n;

suite
  .add("bigint init with hex", function () {
    if (100n > TWO_POW256_BIGINT_HEX) return true;
  })
  .add("bigint init with exp", function () {
    if (100n > TWO_POW256_BIGINT_EXP) return true;
  })
```

Much faster 😄

```
bigint init with hex x 110,702,702 ops/sec ±0.12% (101 runs sampled)
bigint init with exp x 110,732,705 ops/sec ±0.12% (94 runs sampled)
Fastest is bigint init with exp,bigint init with hex
```

One last comparison, with requiring but only once in the file header (snippet):

```js
const { TWO_POW256_BIGINT_HEX, TWO_POW256_BIGINT_EXP } = require("./constants");

suite
  .add("bigint init with hex", function () {
    if (100n > TWO_POW256_BIGINT_HEX) return true;
  })
  .add("bigint init with exp", function () {
    if (100n > TWO_POW256_BIGINT_EXP) return true;
  })
```

Results:
```
bigint init with hex x 85,243,817 ops/sec ±0.41% (90 runs sampled)
bigint init with exp x 86,028,773 ops/sec ±0.19% (97 runs sampled)
Fastest is bigint init with exp
```